Shared/Action/DataStoreAction.swift

```diff
@@ -16,6 +16,59 @@ enum DataStoreAction: Action {
     case sync
     case touch(id: String)
     case delete(id: String)
+    case syncEnd
+    case syncTimeout(error: String)
+    case syncError(error: String)
+}
+
+extension DataStoreAction: TelemetryAction {
+    var eventMethod: TelemetryEventMethod {
+        switch(self) {
+        case .updateCredentials:
+            return .update_credentials
+        case .lock:
+            return .lock
+        case .unlock:
+            return .unlock
+        case .reset:
+            return .reset
+        case .sync:
+            return .sync
```

Comment:
given that there is also sync timeout end and error, should this be sync start or similar?